pkg/cloud/aws/services/kubeadm/aws_defaults.go

```diff
@@ -162,6 +166,11 @@ func SetJoinNodeConfigurationOverrides(caCertHash, bootstrapToken string, machin
 		klog.Infof("Overriding node's cloud-provider to the required value of %q.", cloudProvider)
 	}
 	base.NodeRegistration.KubeletExtraArgs["cloud-provider"] = cloudProvider
+	if machine.Machine.Labels["set"] == "node" {
```

Comment:
Instead of using labels here, this should probably filter on the existence or absence of `machine.spec.versions.controlPlane`.